firebaseai/LiveAudioExample/Audio/AudioPlayer.swift

```
@@ -0,0 +1,85 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AVFoundation
+import Foundation
+
+/// Plays back audio through the primary output device.
+class AudioPlayer {
+  private let engine: AVAudioEngine
+  private let inputFormat: AVAudioFormat
+  private let outputFormat: AVAudioFormat
+  private let playbackNode: AVAudioPlayerNode
+  private var formatConverter: AVAudioConverter
+
+  init(engine: AVAudioEngine, inputFormat: AVAudioFormat, outputFormat: AVAudioFormat) {
+    self.engine = engine
+
+    guard let formatConverter = AVAudioConverter(from: inputFormat, to: outputFormat) else {
+      fatalError("Failed to create the audio converter")
```

Same here - it'd be better to raise an error and handle it in the UI.